internals/daemon/daemon.go

```diff
@@ -47,13 +45,40 @@ import (
 	"github.com/canonical/pebble/internals/systemd"
 )
 
+// notifySupervisor interface defines the required API for
+// sending notifications to a supervisor daemon.
+type notifySupervisor interface {
+	Available() bool
+	Send(state string) error
+}
+
+// shutdown defines the required API for reboot, poweroff
+// and halt control of the underlying system
+type shutdown interface {
```

Comment:
Because external packages need to implement these interfaces (and might want to pass around the type), shouldn't they be exported/capitalised?